README.md

```diff
@@ -1 +1,140 @@
+
+###### The name should always end with what the object is.
+For example: a button must always end with `Button`.
+ * `MenuButton` a button used in the menu
+ * `ButtonMenu` a menu with buttons
+ * `MenuButtonIcon` an icon in the button of the menu
+ 
+There is no limit for the length of a name, so prefer a long name which is clear and descriptive than a short name which is not clear.
+
+#### Abbreviations
+Avoid them as a general rule. For example, `calculateOptimalValue()` is a better method name than `calcOptVal()`. Being clear is more important than minimizing keystrokes. If you don't abbreviate, developers won't have to remember whether you shortened a word like _qualified_ to `qual` or `qlfd`.
+
+Abbreviations that are part of the HTML, CSS and/or JavaScript API are allowed, like:
+ * `URL` for Uniform Resource Locator
+ * `URI` for Uniform Resource Identifier
+ * `src` for source
+ * `DOM` for Document Object Model
+ * `img` for image
+ 
+You should only use these abbreviations within the same context. So only use `img` if you refer to an HTML Image Tag. (`<img>`)
+
+We have standardized on a few abbreviations that are allowed to use:
+ * `app` for application
+ * `args` for arguments
+ * `auto` for automatic, as in `autoLayout`
+ * `bin` for binary
+ * `fps` for frames per second
+ * `id` for identifier. Please note that 'd' should be written in lowercase when used in combination with an other word, like `userId`.
+ * `info` for information, as in `GridRowInfo`
+ * `init` for initialize
+ * `lib` for library
+ * `max` for maximum, as in `maxHeight`
+ * `min` for minimum, as in `minWidth`
+ * `param` for parameter
+ * `params` for parameters
+ * `prop` or `props` for property or properties
+ * `ref` for reference
+ * `temp` for temporary
+ * `ui` for user interface
+ * `util` or `utils` for utility or utilities, as in `StringUtils`
+
+#### Plural or singular?
+
+##### Classes, Interfaces, Types and Enums
+Should always have a **singular** name, unless the object is only used to hold other values and these other value are more important then the object itself, like `Props`, `Settings` or `Options`. For example: `MyComponentProps`, `ProductionSettings` or `CalendarOptions`.
+
+##### Arrays
+Or other kind of lists should have a plural name or end with `List` or `Collection`, like `userList`.
+
+##### Folders
+If a folder holds multiple files, but all related to one main type, it should have a singular name. If it holds multiple main files of a type, it should have a plural name.
+
+For example, the folder `page` contains a single page, with maybe some helper files. The folder `pages` contains multiple pages.   
+
+#### Functions
+Prefer using a verb as a name to indicate it will do something. Like `render` or `open`.
+
+#### Classes, variables, properties, etc.
+All non-functions should have a noun as a name, not a verb.
+
+#### Booleans
+Should start with `is`, `has`, `will` or `should`. Like `isValid` or `hasValues`.
```

Comment:
With https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md it is possible to enforce this with a linting rule. However I noticed that this will cause a lot of issues especially with other libraries.